framework/logging.cpp

```diff
@@ -2308,6 +2282,39 @@ int YamlLogger::print_test_knobs(int fd, mmap_region r)
     return print_count;
 }
 
+void YamlLogger::format_and_print_message(int fd, int message_level, std::string_view message)
+{
+    const char *levels[] = { "error", "warning", "info", "debug" };
+    assert(size_t(message_level) < std::size(levels));
+
+    if (message.find('\n') != std::string_view::npos) {
+        /* multi line */
+        // trim a trailing newline, if any (just one)
+        if (message[message.size() - 1] == '\n')
+            message.remove_suffix(1);
+
+        writeln(fd, indent_spaces(), "    - level: ", levels[message_level]);
+        writeln(fd, indent_spaces(), "      text: |1");
+        print_content_indented(fd, "       ", message);
+    } else {
+        /* single line */
+        writevec(fd, indent_spaces(), "    - { level: ", levels[message_level]);
+        print_content_single_line(fd, ", text: '", message, "' }");
+    }
+}
+
+void YamlLogger::format_and_print_skip_message(int fd, std::string_view message)
+{
+    if (message.find('\n') != std::string_view::npos) {
+        /* multi line */
```

Comment:
Shouldn't this require a trimming of newline here?

```
// trim a trailing newline, if any (just one)
if (message[message.size() - 1] == '\n')
      message.remove_suffix(1);
```

Reply:
Yes and no. The regular message does it because we had old code that had newline, ported over from `printf` logging. We don't have bad `log_skip`.